controllers/provider-aws/pkg/controlplane/actuator.go

```diff
@@ -0,0 +1,363 @@
+	a.chartApplier, err = gardenerkubernetes.NewChartApplierForConfig(config)
+	if err != nil {
+		return errors.Wrap(err, "could not create chart applier")
+	}
+
+	return nil
+}
+
+// InjectClient injects the given client into the actuator.
+func (a *actuator) InjectClient(client client.Client) error {
+	a.client = client
+	return nil
+}
+
+// Reconcile reconciles the given controlplane and cluster, creating or updating the additional Shoot
+// control plane components as needed.
+func (a *actuator) Reconcile(
+	ctx context.Context,
+	cp *extensionsv1alpha1.ControlPlane,
+	cluster *extensionscontroller.Cluster,
+) error {
+	// Decode providerConfig
+	cpConfig := &apisaws.ControlPlaneConfig{}
+	if _, _, err := a.decoder.Decode(cp.Spec.ProviderConfig.Raw, nil, cpConfig); err != nil {
+		return errors.Wrapf(err, "could not decode providerConfig of controlplane '%s'", objectName(cp))
+	}
+
+	// Decode infrastructureProviderStatus
+	infraStatus := &apisaws.InfrastructureStatus{}
+	if _, _, err := a.decoder.Decode(cp.Spec.InfrastructureProviderStatus.Raw, nil, infraStatus); err != nil {
+		return errors.Wrapf(err, "could not decode infrastructureProviderStatus of controlplane '%s'", objectName(cp))
+	}
+
+	// Deploy secrets
+	a.logger.Info("Deploying secrets", "controlplane", objectName(cp))
+	deployedSecrets, err := controlPlaneSecrets.Deploy(a.clientset, a.gardenerClientset, cp.Namespace)
+	if err != nil {
+		return errors.Wrapf(err, "could not deploy secrets for controlplane '%s'", objectName(cp))
+	}
+
+	// Determine subnet ID and zone
+	subnetID, zone, err := getSubnetIDAndZone(infraStatus)
+	if err != nil {
+		return errors.Wrapf(err, "could not determine subnet ID or zone from infrastructureProviderStatus of controlplane '%s'", objectName(cp))
+	}
+
+	// Collect additional configuration chart values
+	values := map[string]interface{}{
+		"cloudProviderConfig": getCloudProviderConfig(infraStatus.VPC.ID, subnetID, zone, cp.Namespace),
+	}
+
+	// Apply config chart
+	a.logger.Info("Applying configuration chart", "controlplane", objectName(cp), "chart", configChart.Name, "values", values)
+	err = configChart.Apply(ctx, a.gardenerClientset, a.chartApplier, cp.Namespace, cluster.Shoot, nil, nil, values)
+	if err != nil {
+		return errors.Wrapf(err, "could not apply configuration chart for controlplane '%s'", objectName(cp))
+	}
+
+	// Compute all needed checksums
+	checksums, err := a.computeChecksums(ctx, deployedSecrets, cp.Namespace)
+	if err != nil {
+		return err
+	}
+
+	// Collect additional CCM chart values
+	values = make(map[string]interface{})
+	if cpConfig.CloudControllerManager != nil {
+		values["featureGates"] = cpConfig.CloudControllerManager.FeatureGates
+	}
+
+	// Apply CCM chart
+	a.logger.Info("Applying CCM chart", "controlplane", objectName(cp), "chart", ccmChart.Name, "values", values)
+	err = ccmChart.Apply(ctx, a.gardenerClientset, a.chartApplier, cp.Namespace, cluster.Shoot, imagevector.ImageVector, checksums, values)
```

Comment:
```suggestion
	if err := ccmChart.Apply(ctx, a.gardenerClientset, a.chartApplier, cp.Namespace, cluster.Shoot, imagevector.ImageVector, checksums, values); err != nil {
			return errors.Wrapf(err, "could not apply CCM chart for controlplane '%s'", objectName(cp))
	}
```

Reply:
Why?

Reply:
Consistency across your code (at some points you are using a one-line if at others you separate), and fewer lines.

Reply:
For consistency across your code, and for shorter lines of code.

Reply:
Done.